pkg/kubelet/status/manager.go

```diff
@@ -332,12 +334,16 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 			m.deletePodStatus(uid)
 			return
 		}
+		if !m.needsUpdate(pod.UID, status) {
```

Comment:
So the reason I moved the check to here, is we need the mirror pod UID. I could look it up in the pod manager, but that would require getting a mutex and either constructing the full name artificially. Moving this method to here means that pod.UID is the UID we need to check (mirror pod, when relevant).

Also, it should be extremely rare that this check ever fails since we only call syncPod if we've already established that an update is needed. The only way this check would fail is if multiple updates were sent in rapid succession. (e.g. status is updated right before syncBatch runs).